app/build.gradle

```diff
@@ -102,13 +102,38 @@ dependencies {
     implementation 'androidx.preference:preference:1.1.1'
     implementation 'com.github.tfcporciuncula:flow-preferences:1.1.0'
 
+    // gRPC
+    implementation("io.grpc:grpc-netty:${grpcVersion}") {
+        exclude module: 'netty-handler'
+    }
+    implementation "io.grpc:grpc-protobuf:${grpcVersion}"
+    implementation "io.grpc:grpc-stub:${grpcVersion}"
+
+    // Android TLS support for Netty
+    implementation('com.github.netty.netty:netty-handler:79ef0c4706') {
+        exclude module: 'netty-common'
+        exclude module: 'netty-resolver'
+        exclude module: 'netty-buffer'
+        exclude module: 'netty-transport'
+        exclude module: 'netty-codec'
+    }
+    // implementation 'io.netty:netty-handler:4.1.48.Final'
+    implementation 'org.conscrypt:conscrypt-android:2.4.0'
+
+    // Base64 encoding
+    // Android comes bundled with 1.3 (but hidden), so we're stuck with this version
```

Comment:
Apologies, I missed this before, but can you please include the URL `https://blog.osom.info/2015/04/commons-codec-on-android.html` in that comment? I'm pretty sure someone (:point_right: me) might try to fiddle with it in the future unless that link is there.